test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java

```diff
@@ -22,115 +22,100 @@
    //Declare things used in the test, like buttons and labels here
    final static int CLICKCOUNT = 10;
    final static int DOUBLE_CLICK_AUTO_DELAY = 10;
-   volatile int lastClickCount = 0;
-   volatile boolean clicked = false;
-   volatile boolean ready = false;
-
-   Frame frame;
-   Robot robot;
-
-   public void init()
-    {
-      //Create instructions for the user here, as well as set up
-      // the environment -- set the layout manager, add buttons,
-      // etc.
-
+   static volatile int lastClickCount = 0;
+   static volatile boolean clicked = false;
+   static volatile boolean ready = false;
+
+   static volatile Frame frame;
+   static volatile Robot robot;
+   static volatile ClickDuringKeypress clicker;
+
+   public static void main(String[] args) throws Exception {
+       clicker = new ClickDuringKeypress();
+       try {
+           EventQueue.invokeAndWait(ClickDuringKeypress::createUI);
+           robot = new Robot();
+           robot.mouseMove(200, 200);
+           EventQueue.invokeAndWait(frame::show);
+           doTest();
+       } finally {
+           if (frame != null) {
+               EventQueue.invokeAndWait(frame::dispose);
+           }
+       }
+   }
+
+   static void createUI() {
       frame = new Frame("ClickDuringKeypress");
-      frame.addMouseListener(this);
+      frame.addMouseListener(clicker);
       frame.addWindowListener(new WindowAdapter() {
           public void windowActivated(WindowEvent e) {
-              synchronized(ClickDuringKeypress.this) {
+              synchronized(clicker) {
                   ready = true;
-                  ClickDuringKeypress.this.notifyAll();
+                  ClickDuringKeypress.clicker.notifyAll();
               }
           }
       });
       frame.setBounds(0, 0, 400, 400);
+    }
 
-      start();
-
-    }//End  init()
-
-   public void start ()
-    {
-      try {
-        robot = new Robot();
-      } catch (AWTException e) {
-        System.out.println("Could not create Robot.");
-        throw new RuntimeException("Couldn't create Robot.  Test fails");
-      }
-
-      robot.mouseMove(200, 200);
-      frame.show();
-
-      synchronized(this) {
-          try {
-              if (!ready) {
-                  wait(10000);
-              }
-          } catch (InterruptedException ex) {
-          }
-          if (!ready) {
-              System.out.println("Not Activated. Test fails");
-              throw new RuntimeException("Not Activated. Test fails");
-          }
+    static void doTest() throws Exception {
+       synchronized(clicker) {
+           try {
+               if (!ready) {
+                   clicker.wait(5000);
+               }
+           } catch (InterruptedException ex) {
+           }
+           if (!ready) {
+               System.out.println("Not Activated. Test fails");
+               throw new RuntimeException("Not Activated. Test fails");
+           }
       }
```

Comment:
Do we really need this synchronization?
Shouldn't the standard 
```java
robot.waitForIdle();
robot.delay(1000);
```
be enough?

Reply:
OK .. changed .. but I'm now a long way from removing an un-used import as a lot of this test is now quite different than before.